var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -289,12 +320,28 @@ def write_library_dirs(f, dirs):
                         )
                     )
 
+    def setup_dependent_build_environment(self, env, depspec):
+        self.setup_build_environment(env)
+
     def setup_build_environment(self, env):
         # Tell numpy which BLAS/LAPACK libraries we want to use.
         # https://github.com/numpy/numpy/pull/13132
         # https://numpy.org/devdocs/user/building.html#accelerated-blas-lapack-libraries
         spec = self.spec
 
```

Comment:
Can you add a note/link for where you found that these env vars are used? They could add new ones in the future.

Reply:
The environment variables come from this file:

https://github.com/numpy/numpy/blob/main/numpy/distutils/system_info.py#L37
```
The order of finding the locations of resources is the following:
 1. environment variable
 2. section in site.cfg
 3. DEFAULT section in site.cfg
 4. System default search paths (see ``default_*`` variables below).
Only the first complete match is returned.
```

The way the detection works, it performs the above scan FOR EACH different type of lapack. That means that if ATLAS is installed in a default location, py-numpy will always prefer it since atlas appears higher in distutils search order.  Adding these environment variables short circuits this search and gives you the correct behavior.